webpage/webstreaming.py

```diff
@@ -0,0 +1,107 @@
+        return jsonify({'results':yAxe})
+
+    @app.route("/save_spectrum", methods=["GET","POST"])
+    def save_spectrum():
+        det._save_spectrum(det.location,det.get_last_spectrum())
+        return "Ok"
+
+    @app.route("/spectroscope", methods=["GET","POST"])
+    def set_config_spectrometer():
+        form = ConfigForm()
+
+        if request.method == 'POST':
+            spectro_scope_config = {}
+            spectro_scope_config['integration_time']     = form.integration_time.data
+            spectro_scope_config['integration_factor']   = form.integration_factor.data
+            spectro_scope_config['threshold']            = form.threshold.data
+
+            set_spectro_scope(app,**spectro_scope_config)
+
+            return redirect(url_for('set_config_spectrometer'))
+
+        else:
+            form.integration_time.render_kw     = {'value':get_spectro_scope('integration_time',app)}
+            form.integration_factor.render_kw   = {'value':get_spectro_scope('integration_factor',app)}
+            form.threshold.render_kw            = {'value':get_spectro_scope('threshold',app)}
+
+            form.integration_time.label         = 'INTEGRATION TIME:'
+            form.integration_factor.label       = 'INTEGRATION FACTOR:'
+            form.threshold.label                = 'THRESHOLD:'
+
+        integration_time                    = get_spectro_scope('integration_time',app)*1000 #Pass integration time to ms
+        det.integration_time    = get_spectro_scope('integration_time',app)
+        det.integration_factor  = get_spectro_scope('integration_factor',app)
+        det.threshold           = get_spectro_scope('threshold',app)
+        return render_template("spectroscope.html",data_x=det.get_wavelengths(),integration_time=integration_time,form=form, auto_en= (True if "automatic" == det.operation_mode else False))
+
+    @app.route("/default",methods=['GET','POST'])
+    def set_default_config():
+        return redirect(url_for('set_config_spectrometer'))
+    return app
+
+def start_webstreaming():
+    ap = argparse.ArgumentParser()
+
+    ap.add_argument("-i", "--ip", type=str, default="0.0.0.0", help="ip address of the device")
+    ap.add_argument("-o", "--port", type=int, default=8081, help="ephemeral port number of the server")
+    ap.add_argument("-l", "--location", type=str, default='captures', help="folder to save the files")
+    ap.add_argument("-id","--ip_det", type=str, default="0.0.0.1", help="ip address of the device")
+
+    args = vars(ap.parse_args())
+
+    # start the flask app
+    print("Using mock spectrometer")
+    spectrometer = MockSpectrometer(ip_address=args["ip_det"], port=1865, channel=0)
+    det = MockDetector(spectrometer)
+    app = create_app(det)
+    init_db(app)
+
+    det.location = args["location"]
+    det.configure_gpio()
+    det.start()
+
+    app.run(host=args["ip"], port=args["port"], debug=True, threaded=True, use_reloader=False)
```

Comment:
@facu-s do you think that we can move this file to root? just like the other project?

Reply:
The principal idea to leave that file inside a specific folder was to have an encapsulated environment for the webpage, thinking on future projects. However I like the idea on having a file at root, working as an interface with the webpage environment.

From my point of view if we are willing to make a change, then we should split the creation of the app of Flask and the definition of it in two different files. Where one should be located inside the webpage folder (the one containing the definition) and the other at root as you say.